### PR TITLE
Adding snapshot and clone test cases of CephFS

### PR DESCRIPTION
**Pull request by AmarnatReddy**
Adding test cases related to Snapshots and Cloning in CEPHFS

Test Cases Covered as part of this PR
	CEPH-83573499 		Remove the original volume after cloning and verify the data is accessible from cloned volume
	CEPH-83573501 		Create a Cloned Volume using a snapshot
	CEPH-83573504 		Verify the status of cloning operation
	CEPH-83573502 		Interrupt the cloning operation in-between and observe the behavior.
	CEPH-83573520 		Validate the max snapshot that can be created under a root FS sub volume level.
				  		Increase by 50 at a time until it reaches the max limit.
	CEPH-83573521 		Remove a subvolume group by retaining the snapshot : ceph fs subvolume rm <vol_n...
	CEPH-83573415 		Test to validate the cli - ceph fs set <fs_name> allow_new_snaps true
	CEPH-83573418 		Create a Snapshot, reboot the node and rollback the snapshot
	CEPH-83573420 		Try writing the data to snap directory
	CEPH-83573524		Ensure the subvolume attributes are retained post clone operations
	CEPH-11319			Create first snap add more data to original then create a second snap.
						Rollback 1st snap do data validation. Rollback 2nd snap and do data validation.
						Perform cross platform rollback i.e. take snap on kernel mount and perform rollback using fuse mount
	CEPH-83573255		Try renaming the snapshot directory and rollbackCreate a FS and
						create 10 directories and mount them on kernel client and fuse client(5 mounts each)
						Add data (~ 100 GB). Create a Snapshot and verify the content in snap directory.
						Try modifying the snapshot name.
	CEPH-83573522		Verify the retained snapshot details with "ceph fs info" command

Logs 5.0 : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4NFP05/

Signed-off-by: Amarnath K <amk@amk.remote.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_utilsV1.py [hunk 1]
```diff
@@ -816,6 +816,176 @@ def get_clone_status(self, client, vol_name, clone_name, **kwargs):
...
+                        "Failed to reconnect to the node {node} after reboot ".format(
+                            node=ceph_node.node.ip_address
+                        )
+                    )
+                    sleep(5)
+                    raise RuntimeError(
+                        "Failed to reconnect to the node {node} after reboot ".format(
+                            node=ceph_node.node.ip_address
+                        )
+                    )
```
Comment (psathyan) [inline tests/cephfs/cephfs_utilsV1.py:910]: We could simplify this loop
```
endtime = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
while datetime.datetime.now() < endtime:
  try:
    node.reconnect()
    return
  except BaseException:
    sleep(5)

raise RuntimeError(...)
```

File: tests/cephfs/snapshot_clone/clone_attributes.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    3. Validate the contents of cloned volume with contents present of subvolume
+    4. Validate file and size xattributes on dir in cloned volume
+
+    Clean Up:
+    1. Delete Cloned volume
+    2. Delete subvolumegroup
+    """
+    try:
+        fs_util = FsUtils(ceph_cluster)
+        config = kw.get("config")
```
Comment (psathyan) [inline tests/cephfs/snapshot_clone/clone_attributes.py:43]: Fail if `config` is not provided... `config = kw["config"]`

File: tests/cephfs/cephfs_utilsV1.py (hunk 1 above)
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_utilsV1.py]: Modified the function and ran the test case which uses it
Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-XXRXRA/

File: tests/cephfs/snapshot_clone/clone_attributes.py (hunk 2 above)
Comment (AmarnatReddy) [reply tests/cephfs/snapshot_clone/clone_attributes.py:43]: By Default we are getting config value from run.py with some details
config["rhbuild"] = f"{rhbuild}-{platform}" if version2 else rhbuild

**Review by psathyan — APPROVED** [review]
Looks good to me

**Pull request merged**